src/population.cpp

```diff
@@ -184,8 +224,27 @@ std::vector<T> Population::getAttribute(const std::string& name, const Selection
 }
 
 
-std::string Population::_attributeDataType(const std::string& name) const
+template <typename T>
+std::vector<T> Population::getEnumeration(const std::string& name, const Selection& selection) const
+{
+    if (impl_->attributeEnumNames.count(name) == 0) {
+        throw SonataError(fmt::format("Invalid enumeration attribute: {}", name));
+    }
+    if (!std::is_integral<T>::value) {
+        throw SonataError(fmt::format("Enumeration attribute '{}' can only be integer", name));
+    }
+
+    HDF5_LOCK_GUARD
+    return _readSelection<T>(impl_->getAttributeDataSet(name), selection);
+}
+
+
+std::string Population::_attributeDataType(const std::string& name, bool remap) const
 {
+    if (remap && impl_->attributeEnumNames.count(name) > 0) {
+        return "string";
```

Comment:
Can we change `remap` to something more descriptive, please?  Maybe something inline with the doc in the .h - `translate_enum_type`?